src/crimson/os/seastore/seastore_types.h

```diff
@@ -411,38 +412,123 @@ class segment_map_t {
  * Fresh extents during a transaction are refered to by
  * record_relative paddrs.
  */
+constexpr uint16_t DEV_ADDR_LEN_BITS = 64 - DEVICE_ID_LEN_BITS;
+enum class addr_types_t : uint8_t {
+  SEGMENT = 0,
+  RANDOM_BLOCK = 1
+};
+using common_addr_t = uint64_t;
```

Comment:
Make this a protected using declaration in paddr_t.  There should be no other references to this type outside of paddr_t and any derived types.